developer/js/source/package-compiler/kmp-compiler.ts

```diff
@@ -71,9 +73,15 @@ export default class KmpCompiler {
 
     // Fill in additional fields
 
-    ['executeProgram', 'graphicFile', 'msiFilename', 'msiOptions', 'readmeFile'].forEach((element) => {
-      if(kps.options[element]) kmp.options[element] = kps.options[element];
-    });
+    let keys: (keyof KpsFileOptions & keyof KmpJsonFileOptions)[] = ['executeProgram', 'graphicFile', 'msiFilename', 'msiOptions', 'readmeFile'];
+    for (let element of keys) {
+      if (kps.options[element]) {
+        if (element === "followKeyboardVersion") {
```

Comment:
I'm a little confused by the need for this. `followKeyboardVersion` is not in the array of `keys` so surely this will never happen?

Reply:
TypeScript's static analysis isn't smart enough to figure out that followKeyboardVersion -- the only non-string property -- is not going to be assigned here. So this was a hack to make the static analysis DEFINITELY not know about it.

But it's a pretty head-scratchy solution, so I decided a type assertion is more clear, along with a comment explaining why.

Reply:
Gotcha. A little bit wild and weird so definitely deserving of a comment! Fun times when we hoopy stuff like this to encourage the compiler to do static analysis properly. It may trip us up if/when we add additional fields to .kps but hopefully we'll be able to see our way through it :)